src/libraries/Microsoft.Extensions.Logging.Abstractions/src/NullLogger.cs

```diff
@@ -20,7 +20,7 @@ private NullLogger()
         }
 
         /// <inheritdoc />
-        public IDisposable BeginScope<TState>(TState state)
+        public IDisposable BeginScope<TState>(TState state) where TState : notnull
```

Comment:
Just curious, how does this not complain about hiding the nullable annotation on ILogger.BeginScope?

Reply:
Your interface implementation can be more specific than the interface. Here the interface says "You can return an IDisposable, or null".  And this implementation says "I only ever return an IDisposable", which is allowed.

The inverse isn't allowed though.